lib/plug/conn.ex

```diff
@@ -445,10 +446,10 @@ defmodule Plug.Conn do
   end
 
   def send_chunked(%Conn{adapter: {adapter, payload}, owner: owner} = conn, status) do
-    conn = run_before_send(%{conn | status: Plug.Conn.Status.code(status), resp_body: nil}, :chunked)
+    conn = run_before_send(%{conn | status: Plug.Conn.Status.code(status), resp_body: nil}, :set_chunked)
```

Comment:
What if we remove the `:set_chunked` state and instead set the response header "transfer-encoding" to chunked. This way we don't need to introduce a new state but we still provide a mechanism for entries inside the before_send callback to check if the response is chunked. What do you think?

Also, no need to rebase. I can rebase when merging!